pkg/apiserver/apiserver.go

```diff
@@ -522,3 +535,99 @@ func (app *uploadAPIApp) createAPIService() error {
 	}
 	return nil
 }
+
+func (app *uploadAPIApp) createWebhook() error {
```

Comment:
Since the apiserver is not upload specific anymore, maybe we should change the name of the struct from uploadAPIApp to something else?  Maybe CDIAPIApp?  But I hate that there are so many acronyms in that!

Reply:
Good idea, the name is confusing and it's indeed not upload specific anymore.
I will open a different issue and make the change.

Reply:
In addition to Michael's comment about the validation, since one of the checks is if the PVC exists, do we need to check if the secret already exists too?